src/apps/dashboard/routes/logs.tsx

```diff
@@ -0,0 +1,150 @@
+import React, { ChangeEvent, FormEvent, useCallback, useEffect, useState } from 'react';
+import type { LogFile } from '@jellyfin/sdk/lib/generated-client/models/log-file';
+import { getConfigurationApi } from '@jellyfin/sdk/lib/utils/api/configuration-api';
+import { getSystemApi } from '@jellyfin/sdk/lib/utils/api/system-api';
+import LogItem from 'components/dashboard/logs/LogItem';
+import Loading from 'components/loading/LoadingComponent';
+import Page from 'components/Page';
+import { useApi } from 'hooks/useApi';
+import globalize from 'lib/globalize';
+import { Alert, Box, Button, FormControlLabel, Stack, Switch, TextField, Typography } from '@mui/material';
+import { Form } from 'react-router-dom';
+
+const Logs = () => {
+    const { api } = useApi();
+    const [ logs, setLogs ] = useState<LogFile[]>([]);
+    const [ logsLoading, setLogsLoading ] = useState<boolean>(true);
+    const [ configLoading, setConfigLoading ] = useState<boolean>(true);
+    const [ logWarningMessageChecked, setLogWarningMessageChecked ] = useState<boolean>(false);
+    const [ slowResponseTime, setSlowResponseTime ] = useState<string>('');
+    const [ submitted, setSubmitted ] = useState<boolean>(false);
+
+    const setLogWarningMessage = useCallback((_: ChangeEvent<HTMLInputElement>, checked: boolean) => {
+        setLogWarningMessageChecked(checked);
+    }, []);
+
+    const onResponseTimeChange = useCallback((event: ChangeEvent<HTMLTextAreaElement>) => {
+        setSlowResponseTime(event.target.value);
+    }, []);
+
+    const loadLogs = useCallback(() => {
+        if (!api) return;
+
+        return getSystemApi(api)
+            .getServerLogs()
+            .then(({ data }) => {
+                setLogs(data);
+            });
+    }, [api]);
+
+    const onSubmit = useCallback((e: FormEvent<HTMLFormElement>) => {
```

Comment:
We should use react router's actions and loaders to initialize the page data and handle submissions. You can see an example of this in https://github.com/jellyfin/jellyfin-web/pull/6334.

Reply:
I've noticed that inside `loader`, the access token for `ServerConnections.getCurrentApi()` will be null when refreshing the page (going from another page to it works fine). This wasn't a problem for the branding page because the configuration request does not need an access token.

Also usage of `loader` will not show a loading indicator.

Have not looked into why that is, but came up with a workaround for now.